Add support for role='alertdialog'.

flu0 opened the pull request:
First time contributor. Posting for feedback before jumping into tests, docs and examples. Hopefully its a small, non-breaking change.

Adds support for role='alertdialog', making the dialog modal by not closing it when the escape key is pressed. Note that the `data-a11y-dialog-hide` attribute should be removed from the overlay as well to comply with the specification.

Closes  #83.

HugoGiraudel commented:
Hey Felipe! Thank you very much for your contribution. That seems to make sense to me. I don’t see any reason not to accept this change. ✨ 

As you mentioned, we need a few more things in order to be able to complete this feature:
- Some tests to make sure it works properly.
- Some documentation to explain how to control the role and when to use which.
- Updating the HTML code snippet about the `data-a11y-dialog-hide` attribute on the overlay.

Thanks again! :)

flu0 commented:
Thanks for the feedback 😄

I'll tackle the docs and examples later today, and try to give a shot at testing. I don't have much experience there so it may take a while.

Quick question: Should I include the minified version on the PR as well or is it generated automatically on merge/release?

HugoGiraudel commented:
@flu0 I’ll take care of version bump + release notes + build. You can focus on the interesting stuff. :)

flu0 commented:
- Added some notes to the readme regarding `alertdialog`, hope it's clear enough. Didn't see the need to include a full blown example on the demo page.
- Added a test for ESC key handling. Is this the right way to do it?

HugoGiraudel commented:
Hey! Thank you, I’ll review it during the week-end. :)

HugoGiraudel commented:
Hey there. I’m coming back to this to see if I can make it happen. I had a question though: does `role="alertdialog"` play nicely with the `<dialog>` element? I know the `<dialog>` element implicitly (when fully supported) add `role="dialog"` to the node. Is it possible to override it with `role="alertdialog"` properly? Is it even legit?

HugoGiraudel commented:
I updated the documentation a little bit. Let me know what you think. :)

flu0 commented:
Thumbs up on the docs, that's a lot cleaner 👍.

About your question, yes, it is legit. Actually, `alertdialog` is the only role allowed on `<dialog>` elements in addition to the default implicit `dialog`, as per the W3C Recommendation: https://www.w3.org/TR/html52/interactive-elements.html#elementdef-dialog

HugoGiraudel commented:
Cool. I’ll wrap this up soon then! ✨

HugoGiraudel commented:
Alright, I’m going to wrap this up today!

HugoGiraudel commented:
Published in 5.2.0, yay!

Thank you so much: https://github.com/edenspiekermann/a11y-dialog/releases/tag/5.2.0. :)

flu0 commented:
Just came back from holidays, thanks for the support and feedback, Hugo!
